lib/iris/fileformats/netcdf/saver.py

```diff
@@ -2455,16 +2474,37 @@ def _lazy_stream_data(data, fill_value, fill_warn, cf_var):
             data = data.squeeze(axis=0)
 
         if is_lazy_data(data):
+            if self.lazy_saves:
+                # deferred lazy streaming
+                def store(data, cf_var, fill_value):
+                    # Create a data-writeable object that we can stream into, which
+                    # encapsulates the file to be opened + variable to be written.
+                    writeable_var_wrapper = (
+                        _thread_safe_nc.DeferredSaveWrapper(
+                            self.filepath, cf_var, self.file_write_lock
+                        )
+                    )
```

Comment:
> reloading cannot really happen in the current form of the code, since the targets that are passed to da.store are actually DeferredSaveWrapper objects

I was thinking in terms of the **_input_** chunks, not the output ones -- the relationship may not always be 1-1 of course.

It's down to dask to only compute each input chunk once.  However, I believe that this is a safe bet, as Dask probably treats the array accesses as generic "tasks", and does not consider its tasks to be re-rerunnable.
So, it is always doing some results caching + re-use.   What is **_not_** so clear, in my experience, is whether it always _discards_ cached results when it could + should.  This is obviously critical to the memory performance of store operations, and I has definitely caused us some problems before.

Reply:
This is a product of black.
So 0 arguments but 0 control over the result !!

I can get rid of the extra parens by making it less wide -- I'll fix the import + naming strategy.